src/prefect/infrastructure/submission.py

```diff
@@ -1,14 +1,23 @@
+import sys
 from typing import Dict, Optional
 
 from anyio.abc import TaskStatus
 
 import prefect
 from prefect.infrastructure.base import Infrastructure
+from prefect.infrastructure.process import Process
 from prefect.orion.schemas.core import FlowRun
 
 MIN_COMPAT_PREFECT_VERSION = "2.0b12"
 
 
+def get_flow_run_entrypoint(infrastructure):
+    if isinstance(infrastructure, Process):
+        return [sys.executable, "-m", "prefect.engine"]
+    else:
+        return ["python", "-m", "prefect.engine"]
```

Comment:
The design of this feels off to me - the location of this command specification is so far removed from where a developer will look when debugging

Reply:
Sad differently, it's far too implicit

Reply:
This single module deals with all of the overrides for running flows in infrastructure. I'm hoping that consolidation here makes it easier to find when debugging in contrast to v1 where you need to look through the implementation details of each agent. I'm not excited about the specific toggle for the Python call in process infrastructure, but I'm not sure where else to implement it. Without the `sys.executable` switch we're too likely to call into the wrong `python` for subprocesses.

Reply:
Are you okay with improving the design in the future? Which user stories are you worried this degrades?

Reply:
This was addressed with a refactor agreed upon in a team review session 🥳